### PR TITLE
Mac System Module

### LINKED ISSUE #30451: create mac_system module
Work with AT command

### PR DESCRIPTION
**Pull request by twangboy**
Added Mac_System module
Added code to System module to Not load on Mac
Fixes #30451

### CODE REVIEW

File: salt/modules/mac_system.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+'''
+System module for sleeping, restarting, and shutting down the system on Mac OS
+X.
+
+Ability to pass times to this module has not been added because the shutdown
+command runs until the time is reached and then shuts the system down. It
+doesn't return until it has acutally started to shut down the system. This
+causes salt CLI to appear to hang because it's waiting for the shutdown command
+to finish. So if you issue `shutdown -r +10`, the shutdown process will run for
+10 minutest and then shutdown the system.
```
Comment (cachedout) [inline salt/modules/mac_system.py]: Could this be handled by piping the shutdown command into `at`?

File: salt/modules/mac_system.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+def _execute_command(cmd):
+    '''
+    Helper function to execute the command
+
+    Returns: bool
+
+    '''
+    ret = __salt__['cmd.retcode'](cmd)
```
Comment (cachedout) [inline salt/modules/mac_system.py]: `return bool(__salt__['cmd.retcode'](cmd))` if you like.

**Comment by cachedout** [comment]
Hi @twangboy . A few comments in-line but no real problems. Have a look and then we can get this merged.

**Comment by twangboy** [comment]
Hold off on Merging this until I get the AT thing working...

**Comment by twangboy** [comment]
@cachedout I got the AT thing to work... If you're good with everything we can merge. Make sure I did the command quoting correctly on line 65

**Comment by cachedout** [comment]
Looks great. As soon as the tests finish, I will merge this.

**Comment by cachedout** [comment]
@twangboy Still some lint errors. Could you please have another look?

**Comment by cachedout** [comment]
@twangboy Could you please fix these lint errors so we can get this in?

**Comment by cachedout** [comment]
Well, now it needs to be rebased too.

**Pull request merged**